lib/resty/rabbitmqstomp.lua

```diff
@@ -116,8 +116,18 @@ function _login(self)
         return nil, err
     end
 
+    local frame, err = _receive_frame(self)
+    if not frame then
+        return nil, err
+    end
+
+    -- We successfully received a frame, but it was an ERROR frame
+    if sub( frame, 1, len( 'ERROR' ) ) == 'ERROR' then
+        return nil, frame
```

Comment:
There isn't an *err* variable in this check - if we reach this line, we've successfully sent a CONNECT frame and received a frame successfully back from the broker. We're looking at the first line of the received frame to see if it starts with the string "ERROR" - if so, this frame is an ERROR frame, which means that we did not connect successfully and that the broker has closed the connection. You can reproduce this by attempting to connect with an invalid username or password.

While we could try to extract the specific error message from the returned frame, as a developer I find that having the entire frame available to me is useful as a debugging tool.